fcrepo-import/src/main/java/org/fcrepo/importer/Importer.java

```diff
@@ -17,26 +17,178 @@
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.io.IOUtils;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ResIterator;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.fcrepo.client.FcrepoClient;
+import org.fcrepo.client.FcrepoOperationFailedException;
+import org.fcrepo.client.FcrepoResponse;
 import org.fcrepo.importexport.TransferProcess;
+import org.slf4j.Logger;
 
 /**
+ * Fedora Import Utility
+ *
  * @author awoods
+ * @author escowles
  * @since 2016-08-29
  */
 public class Importer implements TransferProcess {
+    private static final Logger logger = getLogger(Importer.class);
+    private Config config;
+    protected FcrepoClient client;
 
     /**
      * Constructor that takes the Import/Export configuration
      *
      * @param config for import
      */
     public Importer(final Config config) {
+        this.config = config;
+        this.client = FcrepoClient.client().build();
     }
 
     /**
      * This method does the import
      */
     public void run() {
         System.out.println("Importing!");
+        importDirectory(config.getDescriptionDirectory());
+    }
+
+    private void importDirectory(final File dir) {
+        for (final File f : dir.listFiles()) {
+            if (f.isDirectory()) {
+                importDirectory(f);
+            } else if (f.isFile()) {
+                importFile(f);
+            }
+        }
+    }
+
+    private void importFile(final File f) {
+        FcrepoResponse response = null;
+        URI uri = null;
+        try {
+            final Model model = createDefaultModel().read(new FileInputStream(f), null,
+                    config.getRdfLanguage());
+            final ResIterator binaryResources = model.listResourcesWithProperty(RDF_TYPE, NON_RDF_SOURCE);
+            if (binaryResources.hasNext()) {
+                uri = new URI(binaryResources.nextResource().getURI());
+                logger.info("Importing binary {}", f.getAbsolutePath());
+                response = importBinary(uri, sanitize(model));
+            } else {
+                uri = uriForFile(f, config.getDescriptionDirectory());
+                logger.info("Importing container {}", f.getAbsolutePath());
+                response = importContainer(uri, sanitize(model));
+            }
+
+            if (response.getStatusCode() > 204) {
+                logger.warn("Error while importing {} ({}): {}",
+                   f.getAbsolutePath(), response.getStatusCode(), IOUtils.toString(response.getBody()));
+            } else {
+                logger.info("Imported {}: {}", f.getAbsolutePath(), uri);
+            }
+        } catch (FcrepoOperationFailedException ex) {
+            logger.warn("Error importing {}: {}", f.getAbsolutePath(), ex.toString());
+            ex.printStackTrace();
```

Comment:
I do not think we will want to `ex.printStackTrace()`. Logging the warning is good.
For this exception and the others below, are these not failure scenarios that should be re-thrown?